### PR TITLE
issue-1247: starters for providers layouts

### PR DESCRIPTION
**Pull request by SergeyHydra**
Fixes #1349

### CODE REVIEW

**Comment by steve-todorov** [comment]
@sbespalov , @carlspring could you guys please review? Thanks!

**Comment by carlspring** [comment]
@SergeyHydra ,

Thanks for these changes! They look great!

Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

@sbespalov , @fuss86 ,

Could you please review this as well? Thanks! :)

**Comment by steve-todorov** [comment]
FYI: I've `rebased` this PR.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/config/Maven2LayoutProviderConfig.java [hunk 1]
```diff
@@ -12,10 +12,12 @@
 import java.nio.file.FileSystem;
 import java.nio.file.spi.FileSystemProvider;
 
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.context.annotation.*;
 import org.springframework.core.env.Environment;
 
 @Configuration
+@ConditionalOnClass(Maven2LayoutProvider.class)
```
Comment (fuss86) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/config/Maven2LayoutProviderConfig.java]: unnecessary

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/config/NpmLayoutProviderConfig.java [hunk 2]
```diff
@@ -20,13 +20,16 @@
...
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Scope;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 @Configuration
+@ConditionalOnClass(NpmLayoutProvider.class)
```
Comment (fuss86) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/config/NpmLayoutProviderConfig.java]: unnecessary

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/config/NugetLayoutProviderConfig.java [hunk 3]
```diff
@@ -15,12 +15,14 @@
...
 import java.nio.file.spi.FileSystemProvider;
 
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Scope;
 
 @Configuration
+@ConditionalOnClass(NugetLayoutProvider.class)
```
Comment (fuss86) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/config/NugetLayoutProviderConfig.java]: unnecessary

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/main/java/org/carlspring/strongbox/config/RawLayoutProviderConfig.java [hunk 4]
```diff
@@ -14,12 +14,14 @@
...
 import java.nio.file.spi.FileSystemProvider;
 
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Scope;
 
 @Configuration
+@ConditionalOnClass(RawLayoutProvider.class)
```
Comment (fuss86) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/main/java/org/carlspring/strongbox/config/RawLayoutProviderConfig.java]: unnecessary

**Comment by sbespalov** [comment]
@SergeyHydra thanks for the PR, looks pretty simple and nice :+1: 

as for `@ConditionalOnClass` then it probably not required as @fuss86 noticed, but I think it would be clearer to keep it. What do you think @fuss86 ?

**Comment by fuss86** [comment]
> as for @ConditionalOnClass then it probably not required as @fuss86 noticed, but I think it would be clearer to keep it. What do you think @fuss86 ?

in my opinion this is pointless since conditional classes are in the same packages as configs

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/config/Maven2LayoutProviderConfig.java (hunk 1 above)
Comment (SergeyHydra) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/config/Maven2LayoutProviderConfig.java]: Done. Remove annotations and add last commit PypiLayoutProvider starter

**Comment by carlspring** [comment]
Thanks to everyone involved!

@SergeyHydra, congratulations on your first pull request! :)

**Comment by fuss86** [comment]
@SergeyHydra, congratulations !

Could you please remove unnecessary imports to `org.springframework.boot.autoconfigure.condition.ConditionalOnClass` ?

**Pull request merged**